### PR TITLE
Add phone log flutter plugin

### PR DESCRIPTION
**Pull request by jiajiabingcheng**
Hi team,

I just added a new plugin under the packages/...

Thanks,
Jiaming

### CODE REVIEW

**Comment by jiajiabingcheng** [comment]
Hi @mehmetf, could please review this pr? Thanks!

File: packages/phone_log/.gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.DS_Store
+.atom/
+.idea
+.packages
+.dart_tool/
+.pub/
+build/
+ios/.generated/
+packages
+pubspec.lock
```
Comment (mehmetf) [inline packages/phone_log/.gitignore:10]: You don't need this file. I just updated the .gitignore file at the root. That should be enough.

File: packages/phone_log/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2018 Jiaming Cheng
```
Comment (mehmetf) [inline packages/phone_log/LICENSE:3]: I believe you need to set this to Google Inc. Not your personal name.

File: packages/phone_log/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+```
+## Usage in Dart
+
+Import the relevant file:
+
+```
+import 'package:phone_log/phone_log.dart';
+```
+
+### Methords
```
Comment (mehmetf) [inline packages/phone_log/README.md:29]: Fix typo.

File: packages/phone_log/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+```dart
+/// Check phone log permission and return a [Future] with the result
+static Future<bool> checkPermission(Permission permission);
```
Comment (mehmetf) [inline packages/phone_log/README.md:32]: Avoid static methods. This makes the plugin files non-mockable.

File: packages/phone_log/example/.gitignore [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.DS_Store
```
Comment (mehmetf) [inline packages/phone_log/example/.gitignore:1]: Remove.

File: packages/phone_log/example/android/.gitignore [hunk 6]
```diff
@@ -0,0 +1,10 @@
+*.iml
```
Comment (mehmetf) [inline packages/phone_log/example/android/.gitignore:1]: Remove.

File: packages/phone_log/example/android/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.phonelogexample">
+
+    <!-- The INTERNET permission is required for development. Specifically,
```
Comment (mehmetf) [inline packages/phone_log/example/android/app/src/main/AndroidManifest.xml:4]: Remove comments.

File: packages/phone_log/example/android/app/src/main/res/drawable/launch_background.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Modify this file to customize your launch splash screen -->
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@android:color/white" />
+
+    <!-- You can insert your own image assets here -->
```
Comment (mehmetf) [inline packages/phone_log/example/android/app/src/main/res/drawable/launch_background.xml:6]: Remove comments.

File: packages/phone_log/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -0,0 +1,6 @@
+#Fri Jun 23 08:50:38 CEST 2017
```
Comment (mehmetf) [inline packages/phone_log/example/android/gradle/wrapper/gradle-wrapper.properties:1]: You only need this file. Revert everything else in this directorly. The updated .gitignore should handle all this.

File: packages/phone_log/example/ios/.gitignore [hunk 10]
```diff
@@ -0,0 +1,44 @@
+.idea/
```
Comment (mehmetf) [inline packages/phone_log/example/ios/.gitignore:1]: Revert.

File: packages/phone_log/ios/.gitignore [hunk 11]
```diff
@@ -0,0 +1,36 @@
+.idea/
```
Comment (mehmetf) [inline packages/phone_log/ios/.gitignore:1]: Revert.

**Review by mehmetf — COMMENTED** [review]
I just edited the .gitignore. I recommend syncing your client and creating a new PR. There's a lot of files in here that should not be part of the repo.

**Comment by jiajiabingcheng** [comment]
Thanks a lot Mehmet. As you suggested I'm closing this one and pulling a new one with addressing all the comments.

**Pull request closed without merging**